File: leetcode/Length of Last Word.cpp
#define F(i, a, b) for(int i = a; i <= b; i++)
#include <vector>
#include <unordered_set>

class Solution {
public:
    int lengthOfLastWord(const char *s) {
        if( s == NULL ){
            return 0;
        }
        int i = 0;
        int sum = 0;
        int max = 0;

        while( s[i] != '\0' ){
            if( s[i] != ' ' ){
                sum++;
                max = sum;
            }
            else{
                sum = 0;
            }
            i++;
        }
        return max;
    }
};

File: LeetCode-Java/Length of Last Word.java




public class Solution {
	public int lengthOfLastWord(String s) {
		int retLen = 0;
		int len = s.length();

		int idx = len -1;
		while( idx >= 0 ){
			if( s.charAt(idx) == ' ' ){
				idx--;
			}else{
				while( idx >= 0 && s.charAt(idx) != ' ' ){
					retLen++;
					idx--;
				}
				break ;
			}
		}

		return retLen; 
	}
}



